server/src/main/java/org/apache/cassandra/sidecar/cluster/CQLSessionProviderImpl.java

```diff
@@ -137,16 +168,30 @@ public synchronized Session get()
                                                                      driverUtils);
             // Prevent spurious reconnects of ignored down nodes on `onUp` events
             QueryOptions queryOptions = new QueryOptions().setReprepareOnUp(false);
-            cluster = Cluster.builder()
-                             .addContactPointsWithPorts(contactPoints)
-                             .withReconnectionPolicy(reconnectionPolicy)
-                             .withoutMetrics()
-                             .withLoadBalancingPolicy(lbp)
-                             .withQueryOptions(queryOptions)
-                             // tests can create a lot of these Cluster objects, to avoid creating HWTs and
-                             // event thread pools for each we have the override
-                             .withNettyOptions(nettyOptions)
-                             .build();
+            Cluster.Builder builder
+            = Cluster.builder()
+                     .addContactPointsWithPorts(contactPoints)
+                     .withReconnectionPolicy(reconnectionPolicy)
+                     .withoutMetrics()
+                     .withLoadBalancingPolicy(lbp)
+                     .withQueryOptions(queryOptions)
+                     // tests can create a lot of these Cluster objects, to avoid creating HWTs and
+                     // event thread pools for each we have the override
+                     .withNettyOptions(nettyOptions);
+
+            if (username != null && password != null)
```

Comment:
Should this be an `or` instead of an `and`?

Reply:
Driver doesn't allow password to be null. So I think `and`